llama-demo-apps/StructuredLlama.ipynb

```diff
@@ -0,0 +1,502 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "e8cba0b6",
+   "metadata": {},
+   "source": [
+    "## This demo shows how to use LangChain's SQLDatabaseChain with Llama2 to query about structured data stored in a SQL DB.  \n",
+    "* As the 2023-24 NBA season is around the corner, we use the NBA roster info saved in a SQLite DB to show you how to ask Llama2 questions about your favorite teams or players. \n",
+    "* Because the SQLDatabaseChain API implementation is still in the langchain_experimental package, you'll see more issues that come with using the cutting edge experimental features, and how we succeed resolving some of the issues but fail on some others."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "33fb3190-59fb-4edd-82dd-f20f6eab3e47",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install langchain replicate langchain_experimental"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "70d6c906-4fc0-4da5-acd9-8743e90f89e8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from langchain.llms import Replicate\n",
+    "from langchain.prompts import PromptTemplate\n",
+    "from langchain.utilities import SQLDatabase\n",
+    "from langchain_experimental.sql import SQLDatabaseChain"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "fa4562d3",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdin",
+     "output_type": "stream",
+     "text": [
+      " ········\n"
+     ]
+    }
+   ],
+   "source": [
+    "from getpass import getpass\n",
+    "import os\n",
+    "\n",
+    "REPLICATE_API_TOKEN = getpass()\n",
+    "os.environ[\"REPLICATE_API_TOKEN\"] = REPLICATE_API_TOKEN"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "9dcd744c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "llama2_13b_chat = \"meta/llama-2-13b-chat:f4e2de70d66816a838a89eeeb621910adffb0dd0baba3976c96980970978018d\"\n",
+    "\n",
+    "llm = Replicate(\n",
+    "    model=llama2_13b_chat,\n",
+    "    model_kwargs={\"temperature\": 0.01, \"top_p\": 1, \"max_new_tokens\":500}\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "70776558",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# to see how the db file was created, check the README file\n",
```

Comment:
which readme file we are referring here?

Reply:
good questions - updated the text to remove the reference to readme